components/widgets/modals/full_screen_modal.jsx

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import CloseIcon from 'components/svg/close_icon';
+
+import './full_screen_modal.scss';
+
+export default class FullScreenModal extends React.Component {
+    static propTypes = {
+        show: PropTypes.bool.isRequired,
+        children: PropTypes.node.isRequired,
+        onClose: PropTypes.func.isRequired,
+    }
+
+    componentDidMount() {
+        document.addEventListener('keydown', this.handleKeypress);
+    }
+
+    componentWillUnmount() {
+        document.removeEventListener('keydown', this.handleKeypress);
+    }
+
+    handleKeypress = (e) => {
+        if (e.key === 'Escape' && this.props.show) {
+            this.close();
+        }
+    }
+
+    close = () => {
+        this.props.onClose();
+    }
+
+    render() {
+        return (
+            <div className={'FullScreenModal' + (this.props.show ? ' show' : '')}>
```

Comment:
Do we have to render the empty div when the modal isn't shown? What if we just return null when `!this.props.show`?

Reply:
It just simplify the animation process, but I would be able to wrap it in a `CSSTransition` component.

Reply:
Using a CSSTransition to build this, the problem is how we deal with the modals in a more general way, so for now, the fadeout animation is not working yet (because the container remove the modal before it is able to show the animation), that will be fixed in a more general way in the future.

Reply:
I don't want to keep bugging you about "this is how react-bootstrap does it", but the way that they do it is by having something like a separate `visible` state, and changing `this.props.show` triggers it to change the state. `this.props.show` doesn't directly affect the render method then.

Reply:
Stills needs to be handled from the outside, we need to give it time to disappear from the redux modals machinery, if not, we end up with the same problem.